extensions/project-descriptor.md

```diff
@@ -7,81 +7,144 @@ A project descriptor is a file that MAY contain configuration for apps, services
-source-url = "<url>"
+documentation-url = "<uri>"
+source-url = "<uri>"
 
-[[project.licenses]]
+[[_.licenses]]
 type = "<string>"
 uri = "<uri>"
 
-[build]
-include = ["<string>"]
-exclude = ["<string>"]
-[[build.buildpacks]]
-id = "<string>"
-version = "<string>"
-uri = "<string>"
-[[build.env]]
-name = "<string>"
-value = "<string>"
-[metadata]
+[_.metadata]
 # additional arbitrary keys allowed
 ```
 
-The following sections describe each part of the schema in detail.
-
-## `[project]`
-
-The top-level `[project]` table MAY contain configuration about the repository, including `id` and `version`. It MAY also include metadata about how it is authored, documented, and version controlled.
-
-The `project.id`
+The top-level `_` table MAY contain configuration about the repository, including `id` and `version`. It MAY also include metadata about how it is authored, documented, and version controlled. It MUST contain `schema-version`  to denote which schema version the descriptor is using.
 
 ```toml
-[project]
+[_]
+schema-version = "<string>"
 id = "<string>"
 name = "<string>"
 version = "<string>"
+authors = ["<string>"]
+documentation-url = "<uri>"
+source-url = "<uri>"
 ```
 
+* `schema-version` - version identifier for the schema of the `_` table and structure of the project descriptor file.
 * `id` - (optional) the machine readable identifier of the project (ex. "com.example.myservice")
 * `name` - (optional) the human readable name of the project (ex. "My Example Service")
 * `version` - (optional) and arbitrary string representing the version of the project
 * `authors` - (optional) the names and/or email addresses of the project's authors
-* `documentation-url` - (optional) a URL to the documentation for the project
+* `documentation-url` - (optional) a URL to the documentation for the project.
 * `source-url` - (optional) a URL to the source code for the project
 
-## `[[project.licenses]]`
+#### `_.licenses` (optional)
+
+This table MAY contain project licenses.
 
-An optional list of project licenses.
+```toml
+[[_.licenses]]
+type = "<string>"
+uri = "<uri>"
+```
 
 * `type` - This MAY use the [SPDX 2.1 license expression](https://spdx.org/spdx-specification-21-web-version), but is not limited to identifiers in the [SPDX Licenses List](https://spdx.org/licenses/).
 * `uri` - If this project is using a nonstandard license, then this key MAY be specified in lieu of or in addition to `type` to point to the license.
 
-## `[build.include]` and `[build.exclude]`
+#### `_.metadata` (optional)
 
-A optional list of files to include in the build (while excluding everything else):
+This is a free form table for users to use as they see fit. The keys in this table are not validated.
 
 ```toml
-[build]
+[_.metadata.foo]
+checksum = "a28a0d7772df1f918da2b1102da4ff35"
+```
+
+
+### `io.buildpacks` (optional)
+
+This is the Cloud Native Buildpacks' section of the project descriptor. The TOML schema is the following:
+
+```
+[io.buildpacks]
```

Comment:
If `_.schema-version` declares the version of `_` and the general structure of the project descriptor what versions the structure of this namespace? Say, for example, we change `builder` to be a `table` instead of a `string`. I'd love to see a `schema-version` here to create a distinct separation between the general file format and our namespace.

Reply:
@jromero this came up in my original RFC, but people decided that we wouldn't change the structure much so we wanted only a single version.

Reply:
We can possibly move forward as-is and _when_ we add `schema-version` it would be considered `0.2` or some variant thereof.

Reply:
I'm fine with this.